qdax/baselines/sac.py

```diff
@@ -376,16 +486,21 @@ def _update_critic(
         """
         # update critic
         random_key, subkey = jax.random.split(random_key)
-        critic_loss, critic_gradient = jax.value_and_grad(self._critic_loss_fn)(
+        critic_loss, critic_gradient = jax.value_and_grad(sac_critic_loss_fn)(
             training_state.critic_params,
-            training_state.policy_params,
-            training_state.target_critic_params,
-            alpha,
+            policy_fn=self._policy.apply,
+            critic_fn=self._critic.apply,
+            parametric_action_distribution=self._parametric_action_distribution,
+            reward_scaling=reward_scaling,
+            discount=discount,
+            policy_params=training_state.policy_params,
+            target_critic_params=training_state.target_critic_params,
+            alpha=jnp.exp(training_state.alpha_params),
             transitions=transitions,
-            random_key=subkey,
+            random_key=random_key,
```

Comment:
Same things here concerning the random key flow. The key should not be used here, we should rather use the `subkey`.

Reply:
done